### ISSUE TITLE
Grad norm became Inf and Loss failed to decrease while training

### ISSUE THREAD

**Issue by zhuxinguang33** [megvii-research/PETR#51]
Hi,
Thanks for sharing so wonderful work! I'm trying to reproduce your work now. But according to log file, grad norm became Inf and training loss failed to converge after at the third
![epoch 3](https://user-images.githubusercontent.com/109491964/190332674-1d72fdbf-551d-4e16-aefa-453dd3b50c1a.png)
![grad error1](https://user-images.githubusercontent.com/109491964/190332685-73ecfc07-d32a-4282-9445-65a14a76632e.png)
![grad error2](https://user-images.githubusercontent.com/109491964/190332687-ddd37e74-f941-4700-bc63-c159e0a16017.png)
 epoch. Few changes of code was made but just data root path.

I used 4 GPUs of GeForce RTX 3090 for training. My environment is below:
CUDA: 11.2
python: 3.6.9
pytorch: 1.9.0
mmcv: 1.4.0
mmdet: 2.24.1
mmsegmentation: 0.20.2
mmdet3d: 0.17.1
tools/dist_train.sh projects/configs/petr/petr_r50dcn_gridmask_p4.py 4 --work-dir work_dirs/petr_r50dcn_gridmask_p4/

The first Inf happened at 4200 iters of first epoch, and after 2 epochs, grad norm was almost Inf and Nan

**Comment by yingfei1016** [comment]
Hi,
You can modify the learning rate to 1e-4.

For 4xRTX 3090, I strongly recommend to set samples_per_gpu=4, lr=4e-4. The result may higher than default setting (about ~2%).

**Comment by zhuxinguang33** [comment]
That really helps me a lot. It can be trained correctly now. Thanks for answering.

**Comment by SerCharles** [comment]
how should i set my lr and samples_per_gpu？i need to train on 4 A6000
